### PR TITLE
implement p-to-bf

### PR DESCRIPTION
**Pull request by strengejacke**
``` r
data(iris)
model <- lm(Petal.Length ~ Sepal.Length + Species, data = iris)
bayestestR::p_to_bf(model)
#>           Parameter            p           BF
#> 1       (Intercept) 1.005180e-11 3.693268e-10
#> 2      Sepal.Length 1.121002e-28 4.118823e-27
#> 3 Speciesversicolor 9.645641e-67 3.544035e-65
#> 4  Speciesvirginica 4.917626e-71 1.806851e-69
```

<sup>Created on 2022-05-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Comment by mattansb** [comment]
Fixed: Now returns BF10 (instead of BF01, which is what the formula in the paper give...)

This paper is essentially trying to approximate the BIC to BF approximation (relaying on how Wald tests are asymptotically LRTs)... 

Here is a demo (using @bwiernik `pipebind`!)

``` r
library(bayestestR)
library(pipebind)
#> Warning: package 'pipebind' was built under R version 4.1.3

m0 <- lm(mpg ~ 1, mtcars)
m1 <- lm(mpg ~ am, mtcars)
m2 <- lm(mpg ~ factor(cyl), mtcars)

bic_to_bf(BIC(m1), denominator = BIC(m0))
#> [1] 222.005
m1 |> 
  parameters::p_value() |> 
  bind(x, x[2, "p"]) |> 
  p_to_bf(n_obs = nrow(mtcars))
#>              p       BF
#> 1 0.0002850207 206.7413

bic_to_bf(BIC(m2), denominator = BIC(m0))
#> [1] 45355714
anova(m2) |> 
  parameters::p_value() |> 
  bind(x, x[1, "p"]) |> 
  p_to_bf(n_obs = nrow(mtcars))
#>              p       BF
#> 1 4.978919e-09 11835011
```

<sup>Created on 2022-05-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by mattansb** [comment]
If we do decide to merge this PR, I hope we don't endorse this down the line by adding these to any outputs.

**Comment by DominiqueMakowski** [comment]
> If we do decide to merge this PR, I hope we don't endorse this down the line by adding these to any outputs.

yes i agree

File: R/p_to_bf.R [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#' @examples
+#' if (requireNamespace("parameters", quietly = TRUE)) {
+#'   data(iris)
+#'   model <- lm(Petal.Length ~ Sepal.Length + Species, data = iris)
+#'   p_to_bf(model)
+#' }
+#'
+#' @return A data frame with the p-values and Bayes factors (against the null).
+#'
+#' @seealso [bic_to_bf()]
```
Comment (bwiernik) [inline R/p_to_bf.R:22]:
```suggestion
#' @seealso [bic_to_bf()] for more accurate approximate Bayes factors.
```

**Review by bwiernik — CHANGES_REQUESTED** [review]
Some suggestions for naming and documentation to provide context and recommended usage

**Comment by DominiqueMakowski** [comment]
tbh i wouldn't call it "pseudo" it's a bit long as a name and you know... what is "pseudo" really, if all approximations are pseudo then we should pseudo a *lot* of stuff ^^

**Comment by bwiernik** [comment]
My thinking is that this is much more of a heuristic approximation than other approximations. And it’s a bit long for a reason 😜

**Comment by mattansb** [comment]
If you read the paper they are using the p-value to approximate the SE to approximate the LRT to approximate for the BIC comparisons that can be used to produce BFs. That is a long chain, with many assumptions...

I think `p_to_pseudo_bf()` is a good compromise...

**Comment by DominiqueMakowski** [comment]
okay i'm convinced

**Comment by DominiqueMakowski** [comment]
we should have merged this while i was onboard with p_to_pseudo_bf, now after looking back I think p_to_bf and put the dicslaimer in the title + docstrings should be enough. I mean people will *at least* check the title of the function ^^ wadayathink?

**Comment by mattansb** [comment]
I mean....... No...

**Comment by strengejacke** [comment]
BF is pseudo, anyway, so `p_to_bf()` includes all necessary information?

**Review by bwiernik — APPROVED** [review]

**Comment by strengejacke** [comment]
I'm open for both suggestion, i.e. renaming to `p_to_pseudo_bf()`, or keep it as it is. @DominiqueMakowski do you plan to merge this before submitting _bayestestR_? I think it's feature complete now.

**Comment by mattansb** [comment]
I vote pseudo 👏

**Comment by DominiqueMakowski** [comment]
I vote for keeping it as is

**Comment by strengejacke** [comment]
image

**Comment by strengejacke** [comment]
@easystats/core-team we need some help here...

**Comment by bwiernik** [comment]
Compromise, leave the name and change the output to say “pseudo-BF”?

**Comment by strengejacke** [comment]
> Compromise, leave the name and change the output to say “pseudo-BF”?

Yeah, that's what I already did in the print method, as table header :-)

**Comment by strengejacke** [comment]
image

**Comment by DominiqueMakowski** [comment]
> Yeah, that's what I already did

Meme lord + forward thinker nice

**Comment by DominiqueMakowski** [comment]
Now let's hope this humble offering pleases @mattansb 🙏

**Comment by bwiernik** [comment]
I'm good with it, so we have a majority 😜

**Comment by mattansb** [comment]
I'll take it

**Pull request merged**